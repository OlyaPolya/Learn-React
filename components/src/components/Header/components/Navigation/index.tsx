import React from 'react';
import { Link } from 'react-router-dom';
import { Link as HeaderLink } from '../../../../types';

type Props = {
  links: HeaderLink[];
};

const Navigation: React.FC<Props> = (props) => (
  <ul className="header-links">
    {props.links.map((nav: HeaderLink) => (
      <li key={nav.title} className="link">
        <Link to={nav.url}>{nav.title}</Link>
      </li>
    ))}
  </ul>
);

export default Navigation;
