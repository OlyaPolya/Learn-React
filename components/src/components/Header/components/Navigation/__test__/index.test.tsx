import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Navigation from '..';
import { MemoryRouter } from 'react-router-dom';
import fixtures from '../../../__test__/fixtures';

describe('<Header/>', () => {
  it('should match snapshot', () => {
    const component = render(
      <MemoryRouter>
        <Navigation {...fixtures.props} />
      </MemoryRouter>
    );

    expect(component.container).toMatchSnapshot();
  });

  it('should click on link', () => {
    const component = render(
      <MemoryRouter>
        <Navigation {...fixtures.props} />
      </MemoryRouter>
    );
    const link = component.getByText('Test2');
    fireEvent.click(link);

    expect(link.getAttribute('href')).toEqual('/test');
  });
});
