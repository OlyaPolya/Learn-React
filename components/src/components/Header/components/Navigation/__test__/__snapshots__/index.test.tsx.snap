// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should match snapshot 1`] = `
<div>
  <ul
    class="header-links"
  >
    <li
      class="link"
    >
      <a
        href="/"
      >
        Test1
      </a>
    </li>
    <li
      class="link"
    >
      <a
        href="/test"
      >
        Test2
      </a>
    </li>
  </ul>
</div>
`;
